optimize/backend/src/main/java/io/camunda/optimize/license/CamundaLicense.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.optimize.license;
+
+import org.camunda.bpm.licensecheck.InvalidLicenseException;
+import org.camunda.bpm.licensecheck.LicenseKey;
+import org.camunda.bpm.licensecheck.LicenseKeyImpl;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CamundaLicense {
+
+  public static final String CAMUNDA_LICENSE_ENV_VAR_KEY = "CAMUNDA_LICENSE_KEY";
+  private static final Logger LOGGER = LoggerFactory.getLogger(CamundaLicense.class);
+  private static final String UNKNOWN_LICENSE_TYPE = "unknown";
+  private boolean isValid;
+  private LicenseType licenseType;
+  private boolean isInitialized;
+
+  // @VisibleForTesting
+  protected CamundaLicense() {}
+
+  public CamundaLicense(final String license) {
+    initializeWithLicense(license);
+  }
+
+  public synchronized boolean isValid() {
+    return isValid;
+  }
+
+  public synchronized LicenseType getLicenseType() {
+    return licenseType;
+  }
+
+  public synchronized void initializeWithLicense(final String license) {
+    if (isInitialized) {
+      return;
+    }
+
+    if (license != null && !license.isBlank()) {
+      isValid = determineLicenseValidity(license);
+      licenseType = getLicenseTypeFromProperty(license);
+    } else {
+      isValid = false;
+      licenseType = LicenseType.UNKNOWN;
+      LOGGER.error(
+          "No license detected when one is expected. Please provide a license through the "
+              + CAMUNDA_LICENSE_ENV_VAR_KEY
+              + " environment variable.");
+    }
+
+    isInitialized = true;
+  }
+
+  /**
+   * Camunda licenses will have a license type property, fetch that out of the license and return
+   * the value
+   *
+   * <p>Self-managed mode is any other possibility. (ex, blank license, prop missing, etc)
```

Comment:
I dont understand this sentence 🤔

Reply:
I reworded the method description here, I think its more clear